timeseries/src/autogluon/timeseries/models/autogluon_tabular/transforms.py

```diff
@@ -46,7 +49,9 @@ def apply_inverse_transform(
 ) -> pd.DataFrame:
     """Apply inverse transformation to a dataframe, converting to GroupedArray if necessary"""
     if isinstance(transform, BaseTargetTransform):
-        return transform.inverse_transform(df=df)
+        inverse_transformed = transform.inverse_transform(df=df)
+        assert isinstance(inverse_transformed, pd.DataFrame)
+        return inverse_transformed
```

Comment:
Nit: Can we accomplish this with some comment like a `# type: pd.DataFrame` comment instead of an assertion ([PEP 484](https://peps.python.org/pep-0484/#type-comments))? As far as I understand, we are only using the `assert` to tell pyright "this is definitely a pd.DataFrame, otherwise this code will explode". Maybe we can just keep the first part without the risk of raising an exception

Reply:
see above about type comments. we actually want to raise an exception. the alternative is a polars DataFrame in this case, which we do not want.

Reply:
I did some background reading, turns out my understanding of PEP 484 was incorrect. Disregard my comment.